compiler/qsc_formatter/src/formatter/tests.rs

```diff
@@ -0,0 +1,307 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use expect_test::{expect, Expect};
+use indoc::indoc;
+
+fn check(input: &str, expect: &Expect) {
+    let actual = super::format(input);
+    expect.assert_debug_eq(&actual);
+}
+
+#[test]
+fn test_formatting() {
+    check(
+        "operation   Foo   ()",
+        &expect![[r#"
+    [
+        Edit {
+            span: Span {
+                lo: 9,
+                hi: 12,
+            },
+            new_text: " ",
+        },
+        Edit {
+            span: Span {
+                lo: 15,
+                hi: 18,
+            },
+            new_text: "",
+        },
+    ]
+"#]],
+    );
+}
+
+#[test]
+fn remove_trailing_spaces() {
```

Comment:
When are trailing spaces removed? Do we need more source states to show that it is getting removed in all cases? Should there be a test per kind of trailing space that gets cleaned up?

Reply:
All trailing spaces are removed. I'm not really sure what you are getting at here. The only exception I can think of is with a multi-line string, as the contents of strings will not be edited. I could write a test to check that that is the case.